src/main/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileMetadataRepository.scala

```diff
@@ -26,4 +26,12 @@ class FileMetadataRepository(db: Database) {
       .length
     db.run(query.result)
   }
+
+  def addChecksumValidationResult(fileId: UUID, validationResult: Option[Boolean]): Future[Int] = {
+    val query = for {
+      file <- File if file.fileid === fileId
+    } yield file.checksummatches
+    val update = query.update(validationResult)
+    db.run(update)
```

Comment:
Is it possible to run this update in the same transaction as saving the metadata? That would avoid the situation where one update succeeds and the other fails, leaving them out of date.

I haven't tried this with Slick before, but it looks like you can add `transactionally` to the end of an action which contains updates to multiple tables: https://scala-slick.org/doc/3.2.0/dbio.html#transactions-and-pinned-sessions

Reply:
Yep. I've changed it so both the insert and update run in the same transaction and I've refactored it a bit so this only happens with a checksum update.